### ISSUE TITLE
Do not receive a RELEASE after opening a file in shell with 'tail' command or with 'Microsoft Powerpoint' and editing and removing the file at the same time

### ISSUE THREAD

**Issue by SajadMirzaei** [osxfuse/osxfuse#488]
## Overview
When the `-o "hard_remove"` is used, removing a file while it is open by other programs (processes) does not generate a `RELEASE` signal. 
I opened a file with 'cat' command and in a separate terminal used the 'tail' command. I edited the file with 'cat' and the update showed up in 'tail'. If I use 'rm' command in a different terminal and remove the file, closing the 'cat' will ctrl+C with send a RELEASE but closing the 'tail' does not. 
Also opening a file with Microsoft Powerpoint, editing the file and closing has the same issue. The number of RELEASE signals received are one less than the OPEN and CREATE signals (Powerpoint at some point deletes the file and creates it right away which is the same scenario as above). Here are the reproducing steps for this issue. 

### 'tail' and 'cat'

- Install osxfuse (3.7.1) (tested on OSX 10.12.6)
- Mount filesystem to a directory (let's assume `<dir>`) using the following command in a terminal window (window 0): `./loopback -f -d -o "hard_remove" <dir>` (I am watching this window for OPEN and RELEASE and etc.)

- In terminal window 1: `touch <test-dir>/file.txt` (The path to a test directory within the mounted directory)
- In terminal window 2: `cat > <test-dir>/file.txt`
- In terminal window 3: `tail -f <test-dir>/file.txt`
- In terminal window 2: enter some text and press enter (changes will show up in window 3)
- In terminal window 1: `rm <test-dir>/file.txt`
- In terminal window 2: `Ctrl-C` to terminate the `cat` command (This will show a RELEASE)
- In terminal window 3: `Ctrl-C` to terminate the `tail` command (This will not show a RELEASE)

### Microsoft Powerpoint
- Install osxfuse (3.7.1) (tested on OSX 10.12.6)
- Mount filesystem to a directory (let's assume `<dir>`) using the following command in a terminal window (window 0): `./loopback -f -d -o "hard_remove" <dir>` (I am watching this window for OPEN and RELEASE and etc.)

- Create or Copy a powerpoint file (let's say file.pptx) in a test directory (let's say `<test-dir>`, the path to a test directory within the mounted directory)
- In a different terminal window: `open -b com.microsoft.PowerPoint <test-dir>/file.pptx`
- Edit the file in Microsoft Powerpoint and exit.

**Comment by jeremyspiegel** [comment]
I created a more minimal reproducible case:
```c
#include <stdio.h>
#include <fcntl.h>
#include <unistd.h>

int main() {
    const char* file = "/path/to/writeable/dir/in/loopback/file.txt";
    int fd1 = creat(file, S_IRUSR|S_IWUSR);
    write(fd1, "a", 1);
    int fd2 = open(file, O_RDWR);
    unlink(file);
    close(fd1);
    close(fd2);
}
```

The release for `fd1` is called with `close(fd1)`, but no release occurs for `close(fd2)`. Also, `close(fd2)` returns 0 (to indicate success).

**Comment by bfleischer** [comment]
Thanks for the very detailed bug report and sample code. The issue should be fixed in version 3.8.1. Can you confirm that the issue has been resolved?

**Comment by jeremyspiegel** [comment]
@bfleischer, I've confirmed that the issue is resolved in 3.8.1. Thanks!
